public/docs/ts/latest/cookbook/third-party-lib.jade

```diff
@@ -0,0 +1,247 @@
+include ../_util-fns
+
+:marked
+  Traditionally, third party JavaScript libraries have been published in the form of a single JavaScript file.
+  Consumers of the library have then included the library, "as is", somewhere on the page using a `script` tag.
+
+  Modern web development has changed this process. Instead of publishing a "one size fits all" bundle, developers want to only include the parts of the library they actually need.
+
+  This cookbook shows how to publish a third party library in a way that makes it possible to take advantage of techniques like Ahead of Time Compilation (AoT) and Tree Shaking.
+
+<a id="toc"></a>
+:marked
+  ## Table of contents
+
+  [Creating a Third Party Library](#third-party-lib)
+
+  [Supporting AoT](#aot)
+
+  [Preparing the library for Tree Shaking](#tree-shaking)
+
+  [Supporting JiT](#jit)
+
+  [Publish](#publish)
+
+  [Integrate with Application](#integrate-with-app)
+
+  [Final Application](#final-app)
+
+.l-main-section
+<a id="third-party-lib"></a>
+:marked
+  ## Creating a Third Party Library
+
+  This cookbook shows how to create a simple Hero-Profile library and publish it with support for AoT compilation and Tree Shaking. 
+  
+  A version of the library intended for JiT compiled applications will also be included. 
+
+  The code for the Hero-Profile library can be found below.
+
+  To support both AoT and JiT compilation there are two different tsconfig files.
+  
+  `tsconfig.json` for JiT compilation and `tsconfig-aot.json` for AoT compilation.
+
++makeTabs(
+  `cb-third-party-lib/hero-profile/hero-profile.module.ts,
+   cb-third-party-lib/hero-profile/tsconfig-aot.json,
+   cb-third-party-lib/ts/tsconfig.json,
```

Comment:
Can you make `tsconfig.1.json` for this chapter with the config you want to show and use it here?

Our internal tsconfigs have some properties that shouldn't be shown to users:
```
    "typeRoots": [
      "../../node_modules/@types/"
    ]
```
```
  "exclude": [
    "node_modules/*",
    "**/*-aot.ts",
    "app-aot"
  ]
```